File: scripts/gauss.rb
def gauss(theMatrix) 
    int i,j,ij,k;
    double error,u[m∗n],z;
    
    # Set initial guess to be identically zero
    for(ij=0;ij<m∗n;ij++) 
        u[ij]=0;
    end
    outputanderror("gsrb out",u,0);
    
    # Compute Red−Black Gauss−Seidel iteration
    for(k=1;k<=total iters;k++)
        
        for(j=1;j<n−1;j++) 
            for(i=1+(j&1);i<m−1;i+=2) 
                16ij=i+m∗j;
                u[ij]=(f(i,j)+dxxinv∗(u[ij−1]+u[ij+1])
                +dyyinv∗(u[ij−m]+u[ij+m]))∗dcent;
            end
        end
        
        for(j=1;j<n−1;j++) 
            for(i=2−(j&1);i<m−1;i+=2) 
                ij=i+m∗j;
                u[ij]=(f(i,j)+dxxinv∗(u[ij−1]+u[ij+1])
                +dyyinv∗(u[ij−m]+u[ij+m]))∗dcent;
            end
        end
        # Save the result and compute error if necessary
        outputanderror("gsrb out",u,k);
    end
end

File: scripts/gauss.lua
function gauss(theMatrix)
    local i,j,ij,k;
    local error,u[m∗n],z;

    -- Set initial guess to be identically zero
    local guessMat = (0 for [0...m*n])
    logError("gsrb out", u, 0)

    -- Compute Red−Black Gauss−Seidel iteration
    for k = 1, totaliters do

        for j = 1, n - 1 do
            for i = (1 + (j and 1)), m - 1, 2 do
                ij = i + m * j
                guessMat[ij] = (f(i,j) + dxxinv * (guessMat[ij - 1] + guessMat[ij + 1]) + dyyinv * (guessMat[ij - m]+guessMat[ij + m])) * dcent;
            end
        end

        for j = 1, n - 1 do
            for i = (1 + (j and 1)), m - 1, 2 do
                ij = i + m * j
                guessMat[ij] = (f(i,j) + dxxinv * (guessMat[ij - 1] + guessMat[ij + 1]) + dyyinv * (guessMat[ij - m] + guessMat[ij + m])) * dcent;
            end
        end

        -- Save the result and compute error if necessary
        logError("gsrb out", u, k)
    end
end
